src/core/thread/mle.cpp

```diff
@@ -458,7 +458,6 @@ otError Mle::Store(void)
         // avoid losing/overwriting previous information when a reboot
         // occurs after a message is sent but before attaching.
 
-        networkInfo.mDeviceMode          = mDeviceMode.Get();
```

Comment:
It looks like this was put in this if statement to fix a previous bug--is it pretty safe to pull it out?

Reply:
I think this is actually key part of the change in behavior we want, i.e. to update the `mDevcieMode` based on current MLE value even if not currently attached - to ensure user-requested changes to `mDeviceMode` (even while device is detached or disabled) are saved and persisted.

In general, when the `Store()` is called the `mDeviceMode` should be already valid (e.g., from `Restore()` which is called right after the ot instance is initialized (and all object constructors are run),  we do read and set the `mDeviceMode` from the saved entry in non-volatile).